### PR TITLE
Document generator creation procedure

### LINKED ISSUE #38: Message Generators Creation Method
Since the consensus seems to be to move to global fixed message generators shared between issuers (see #19 ) the next step would be to decide how to create those generators.  Opening this issue to create some discussion around possible proposals. My understanding is that they will be deterministically generated by a common seed (created from some nothing-up-my-sleeve value).

I would also propose for the process to be such that it will create a strong binding between index and generator (i.e., something like `hash-to-curve(seed, dst, index)`) to help with indexing in applications and avoiding ambiguities.

### PR DESCRIPTION
**Pull request by tplooker**
Initial stab at this attempting to document and resolve #38. In general I was keen to define a generic procedure that could take a seed scoped either globally, issuer or signature specific. Detail about the concrete seed we are using the BLS12-381 cipher suite is still missing.

### CODE REVIEW

File: draft-bbs-signatures.md [hunk 1]
```diff
@@ -556,6 +555,34 @@ Procedure:
...
+
+Outputs:
+
+- generators, an array of generators
+
+Procedure:
+
+1. h = XOF(seed)
+
+2. for i in 0 to length: generators &= hash_to_curve_g1(h.read(64), dst)
```
Comment (andrewwhitehead) [inline draft-bbs-signatures.md:581]: Maybe `generators_i =`?

File: draft-bbs-signatures.md [hunk 2]
```diff
@@ -137,11 +140,7 @@ P1, P2
...
-algorithm, once selected apply the isogeny simplified SWU map to compute a point in G1 using the random oracle method. The domain separation tag value is dst.
-
-hash\_to\_curve\_g2(ostr) -> P
-: The cryptographic hash function that takes as an arbitrary octet string input and returns a point in G2 as defined in [@!I-D.irtf-cfrg-hash-to-curve]. The algorithm first requires selection of the pairing friendly curve and digest algorithm, once selected the isogeny simplified SWU map to compute a point in G2 using the random oracle method. The domain separation tag value is dst.
+: The cryptographic hash function that takes as an arbitrary octet string input and returns a point in G1 as defined in [@!I-D.irtf-cfrg-hash-to-curve]. The algorithm first requires selection of the pairing friendly curve and digest algorithm, once selected the isogeny simplified SWU map to compute a point in G1 using the random oracle method. The domain separation tag value is dst.
```
Comment (BasileiosKal) [inline draft-bbs-signatures.md:143]:
```suggestion
: The cryptographic hash function that takes as an arbitrary octet string input and returns a point in G1 as defined in [@!I-D.irtf-cfrg-hash-to-curve]. The algorithm first requires selection of the pairing friendly curve and digest algorithm. Once selected, apply the isogeny simplified SWU map to compute a point in G1 using the random oracle method. The domain separation tag value is dst.
```

Comment (BasileiosKal) [reply draft-bbs-signatures.md:143]: Very minor suggestion.

File: draft-bbs-signatures.md [hunk 3]
```diff
@@ -137,11 +140,7 @@ P1, P2
 : points on G1 and G2 respectively. For a pairing-friendly curve, this document denotes operations in E1 and E2 in additive notation, i.e., P + Q denotes point addition and x \* P denotes scalar multiplication. Operations in GT are written in multiplicative notation, i.e., a \* b is field multiplication.
 
 hash\_to\_curve\_g1(ostr) -> P
-: The cryptographic hash function that takes as an arbitrary octet string input and returns a point in G1 as defined in [@!I-D.irtf-cfrg-hash-to-curve]. The algorithm first requires selection of the pairing friendly curve and digest
```
Comment (mikelodder7) [inline draft-bbs-signatures.md:140]: I think this is too long. I would just leave it as 

```
The cryptographic hash function that takes as an arbitrary octet string input and returns a point in G1 as defined in [@!I-D.irtf-cfrg-hash-to-curve].`

File: draft-bbs-signatures.md [hunk 4]
```diff
@@ -137,11 +140,7 @@ P1, P2
...
-algorithm, once selected apply the isogeny simplified SWU map to compute a point in G1 using the random oracle method. The domain separation tag value is dst.
-
-hash\_to\_curve\_g2(ostr) -> P
-: The cryptographic hash function that takes as an arbitrary octet string input and returns a point in G2 as defined in [@!I-D.irtf-cfrg-hash-to-curve]. The algorithm first requires selection of the pairing friendly curve and digest algorithm, once selected the isogeny simplified SWU map to compute a point in G2 using the random oracle method. The domain separation tag value is dst.
+: The cryptographic hash function that takes as an arbitrary octet string input and returns a point in G1 as defined in [@!I-D.irtf-cfrg-hash-to-curve]. The algorithm first requires selection of the pairing friendly curve and digest algorithm. Once selected, apply the isogeny simplified SWU map to compute a point in G1 using the random oracle method. The domain separation tag value is dst.
```
Comment (tplooker) [inline draft-bbs-signatures.md:143]:
```suggestion
: The cryptographic hash function that takes as an arbitrary octet string input and returns a point in G1 as defined in [@!I-D.irtf-cfrg-hash-to-curve].
```

As per @mikelodder7 suggestion

File: draft-bbs-signatures.md (hunk 1 above)
Comment (tplooker) [reply draft-bbs-signatures.md:581]: With the merging of #64 agree

Comment (tplooker) [reply draft-bbs-signatures.md:581]:
```suggestion
2. for i in 0 to length: generators_i = hash_to_curve_g1(h.read(64), dst)
```

**Review by mikelodder7 — APPROVED** [review]

**Review by BasileiosKal — APPROVED** [review]

**Comment by tplooker** [comment]
Multiple approvals, open for two weeks, merging.

**Pull request merged**